### ISSUE TITLE
UnknownAdapter and node_modules

### ISSUE THREAD

**Issue by isuvorov** [balderdashy/sails#3006]
I use Docker structure like this https://github.com/b00giZm/docker-compose-nodejs-examples/tree/master/05-nginx-express-redis-nodemon

/src/package.json - copy of package.json
/src/node_modules - all dependencies
/src/app  -  Sails app

All work fine, but there some trouble:

https://github.com/balderdashy/sails/blob/74a822aa6744cce9600b65f21bbd8f087413bbe8/lib/hooks/orm/backwards-compatibility/upgrade-datastore.js
// Before trying to actually require the adapter, make sure we know the real module path:
var node_modules = path.resolve(sails.config.appPath, 'node_modules');

I think, Sails try to find adapter in /src/app/node_modules
How can it fix?

**Comment by jdcauley** [comment]
@isuvorov 

Why do you have your node_modules in an alternate directory? Out of curiosity.

I may be wrong but you may be able to define this in the sailsrc file

**Comment by kevinburkeshyp** [comment]
This is a dupe of https://github.com/balderdashy/sails/issues/2505, since module paths are hard coded in Sails there's currently no way around this besides symlinking the node_modules folder into the "right" place.

**Comment by sailsbot** [comment]
Thanks for posting, @isuvorov.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
